drivers/pwm/pwm_nrfx.c

```diff
@@ -12,86 +12,245 @@ LOG_MODULE_REGISTER(pwm_nrfx);
 
 #define PWM_NRFX_CH_VALUE_NORMAL        (1UL << 15)
 #define PWM_NRFX_CH_VALUE_INVERTED      (0)
+#define NRFX_PWM_PIN_INVERTED_MASK      0x80
+#define NRFX_PWM_PIN_MASK               0x7F
+#define NRFX_PWM_POLARITY_MSK           0x8000
+#define NRFX_PWM_COMPARE_MASK           0x7FFF
 
 struct pwm_nrfx_config {
 	nrfx_pwm_t pwm;
-	nrfx_pwm_config_t config;
-	nrf_pwm_sequence_t seq;
 };
 
 struct pwm_nrfx_data {
+	nrfx_pwm_config_t config;
```

Comment:
That can of course be done if desired. I just feel like placing this structure under pwm_nrfx_config, and then using pwm_nrfx_config as a const struct under DEVICE_DEFINE might be confusing since we in fact are changing some of the members in it when its used; but i have no strong opinions on this, if you want me to change it back and add pointers in the data struct i can do that.

Another point i would like to make is that i foresee us being able to change the PWM pins in the future. There is nothing stopping us from changing the PWM output GPIOs in runtime when none of the channels are in use; so that leaves even more of the config struct changing at run-time?

Reply:
As discussed, moved configuration back to config struct.